src/org/zaproxy/zap/extension/spider/SpiderAPI.java

```diff
@@ -460,6 +468,33 @@ public ApiResponse handleApiView(String name, JSONObject params) throws ApiExcep
 				resultList.addItem(new ApiResponseSet("scan", map));
 			}
 			result = resultList;
+		} else if (VIEW_ALL_URLS.equals(name)) {
+			ApiResponseList resultUrls = new ApiResponseList(name);
+			Set<String> urlSet=new HashSet<String>();
+			
+			TableHistory tableHistory = extension.getModel().getDb().getTableHistory();
+			List<Integer> ids = Collections.emptyList();
+			
+			try {
+				ids = tableHistory.getHistoryIdsOfHistType(extension.getModel().getSession().getSessionId(),
+						HistoryReference.TYPE_SPIDER, HistoryReference.TYPE_SPIDER_TASK);
+			} catch (DatabaseException e) {
+				throw new ApiException(ApiException.Type.INTERNAL_ERROR, e.getMessage());
+			}
+			
+			String url;
+			for (Integer id : ids) {
+				try {
+					url = tableHistory.read(id.intValue()).getHttpMessage().getRequestHeader().getURI().toString();
+					if (urlSet.add(url)) {
+						resultUrls.addItem(new ApiResponseElement("url", url));
+					}
+				} catch (HttpMalformedHeaderException | DatabaseException e) {
+					throw new ApiException(ApiException.Type.INTERNAL_ERROR, e.getMessage());
+				}
+			}
+			
```

Comment:
Actually, noticed a potential issue, we should also test that the `RecordHistory` is not `null` before using it, that can happen if one of the entries is deleted while we are iterating the IDs (likely to happen for `TYPE_SPIDER_TASK`, since those are deleted once no longer needed).